lib/monotime.h

```diff
@@ -59,6 +59,17 @@ struct printfrr_eargs;
 	} while (0)
 #endif
 
+static inline uint64_t monotime_nano(void)
+{
+	struct timespec ts;
+	struct timeval tvo;
+
+	clock_gettime(CLOCK_MONOTONIC, &ts);
+	TIMESPEC_TO_TIMEVAL(&tvo, &ts);
+	return tvo.tv_sec * 1000000L + tvo.tv_usec;
```

Comment:
This `nano` function is returning `micro` seconds.

Also why not just use `struct timeval?`

Reply:
it was originally nano but not all platforms return that :(.  I'll fix the name.  struct timeval is much larger and was hoping to not grow the data structure

Reply:
Memory impact here is 1 per `route_entry`, so a few million worst-case? That'd be 20-30 MB. (Again, worst case.)

Balancing against that, we have the cost of adding _yet another_ time datatype.

I'm really torn about this TBH. 20-30 MB isn't negligible, but it's also far less than what we waste in other places. And I was hoping to cut down the number of timekeeping approaches we have, not increase them…

To get me to "neutral" on this, it'd need to be "best we can do" for this type. That means:

- make it nanoseconds, platforms that don't support this just get trailing `000`.
- add `printfrr` format specifier to print this, cf. `%pTT` / `%pTV` / `%pTS`
- this really needs to be a distinct type. I originally thought that means a `nsec64_t` typedef. But actually — I think a single-member struct would be even better, e.g.:

```
struct monotime_ns {
	uint64_t ns;
};
typedef struct monotime_ns monotime_ns;
```

The reason for making it a struct like this is extremely simple: the compiler won't let you mix it with other random numbers. Typedefs are weak types, structs are strong types. `time_t` already exists as a typedef. Passing a `time_t` into a `uint64_t` won't even get you a raised eyebrow from the compiler. Trying to pass a `time_t` into `[struct] monotime_ns` will get you an immediate "NO U" from the compiler. And mixing up time types has a huge potential for fuckups.

(NB: This struct should be passed by value, not by pointer. General C ABI for a single-member struct means that the member is passed as if it were a straight value.)

Also — again, just to be clear, those comments get me to "neutral" on this. I'm still not sure whether going with `timeval` isn't the better choice.